code/game/objects/items/stacks/medical.dm

```diff
@@ -77,6 +78,41 @@
 							 "<span class='green'>You apply [src] on [M].</span>")
 		use(1)
 
+/obj/item/stack/medical/proc/heal(mob/living/M, mob/user)
+	var/mob/living/carbon/human/H = M
+	var/obj/item/organ/external/affecting = H.get_organ(user.zone_sel.selecting)
+	user.visible_message("<span class='green'>[user] [healverb]s the wounds on [H]'s [affecting.name].</span>", \
+						 "<span class='green'>You [healverb] the wounds on [H]'s [affecting.name].</span>" )
+
+	var/rembrute = max(0, heal_brute - affecting.brute_dam)
+	var/remburn = max(0, heal_burn - affecting.burn_dam)
+	var/nrembrute = rembrute
+	var/nremburn	= remburn
+	affecting.heal_damage(heal_brute, heal_burn)
+	var/list/achildlist = affecting.children.Copy()
+	var/parenthealed = 0
+	while(rembrute + remburn > 0): // Don't bother if there's not enough leftover heal
+		var/obj/item/organ/external/E
+		if(achildlist.len):
+			E = pick(achildlist) // Pick a random children
+			achildlist -= E
+		else if(affecting.parent && !parenthealed): // If there's a parent and no healing attempt was made on it
+			E = affecting.parent
+			parenthealed = 1
+		else:
+			break // If the organ have no child left and no parent / parent healed, break
+		if(E.status & ORGAN_ROBOT || E.open) // Ignore robotic or open limb
+			continue
+		else if(!E.brute_dam && !E.burn_dam) // Ignore undamaged limb
+			continue
+		nrembrute = rembrute - E.brute_dam // Deduct the healed damage from the remain

```

Comment:
Should these values be `max`ed with 0

Reply:
They're maxed to zero since heal_damage doesn't have a safety check for negative values. You can pass in a negative value and have it deal damage. 

	brute_dam = max(brute_dam - brute, 0)
	burn_dam  = max(burn_dam - burn, 0)

Reply:
So if you had a lower body with 5 points of brute damage, and a leg with 30 points of burn damage, and another leg with 10 points of brute
and then you used a brute kit with 25 points
You'd heal the lower body, `25-5 = 20` points of overflow
You check the leg with 30 burn
`nrembrute = 20 - 0`
`nremburn = 0 - 30`
Then you hit the `while` check again, `20 + (-30) > 0 == FALSE`
So the loop terminates, and you don't get to hit the leg with 10 brute